### PR TITLE
Fixes pkg/kubectl/cmd/auth/reconcile.go Remove dependency

### PR DESCRIPTION
**Pull request by shiywang**
Fixes https://github.com/kubernetes/kubectl/issues/91

~~~this maybe an unsafe bump, I will first test if it pass or not.~~~  deepcopy generate done

cc @mengqiy @sttts @liggitt @totherme

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/auth/reconcile.go [hunk 1]
```diff
@@ -161,7 +160,7 @@ func (o *ReconcileOptions) RunReconcile() error {
 		shallowInfoCopy := *info
 
 		switch t := info.Object.(type) {
-		case *rbac.Role:
+		case *rbacv1.Role:
```
Comment (liggitt) [inline pkg/kubectl/cmd/auth/reconcile.go]: this means you can no longer run `kubectl auth reconcile -f <file containing v1beta1 RBAC objects>`

**Comment by liggitt** [comment]
cc @kubernetes/sig-cli-pr-reviews @kubernetes/sig-auth-pr-reviews

**Comment by deads2k** [comment]
omg.  Are we really sure this cure isn't worse than the disease?  Adding 6k lines (just here!) doesn't really feel like the next best solution.

**Comment by liggitt** [comment]
agree, this doesn't seem maintainable

**Comment by shiywang** [comment]
@liggitt @deads2k yes, I agree this is really huge.. the good news is for now those three bump file commits `v1` `v1alpha1` `v1beta1` are all same only expect the version imported, if this reconciliation logic not going to vastly different among versions, I think maybe we can make those three packages of code generated ? 

But if that different version and logic gonna change dramatically in the future, I agree this doesn't seem maintainable.

@pwittrock @mengqiy wdyt ?  have any better ideas ?

**Comment by liggitt** [comment]
> if this reconciliation logic not going to vastly different among versions, I think maybe we can make those three packages of code generated?

This seems like doubling down on a bad direction. Writing logic once and using it against all supported versions of a resource is a *very* common goal. I don't think it should require writing shim interfaces and generating code on top of API objects.

**Comment by ericchiang** [comment]
Removing this from the 1.10 milestone since it hasn't been active for 3 months.

**Comment by liggitt** [comment]
reconcile moved to use external APIs in 1.11

/close

**Pull request closed without merging**